internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go

```diff
@@ -1162,14 +1162,14 @@ func (h *Handler) listPublicShares(r *http.Request, filters []*link.ListPublicSh
 			}
 
 			statResponse, err := c.Stat(ctx, statRequest)
-			if err != nil {
-				return nil, err
+			if statResponse.Status.Code != rpc.Code_CODE_OK {
+				if statResponse.Status.Code == rpc.Code_CODE_NOT_FOUND {
```

Comment:
is status code is neither or nor not found, should still return `nil, err`, might need to format that error